src/main/index.js

```diff
@@ -138,3 +143,86 @@ autoUpdater.on('update-downloaded', (info) => {
   mainWindow.webContents.send("updateDownloaded");
   autoUpdater.quitAndInstall()
 });
+
+
+/**
+ * Local server to listen for redirect uri for Oauth2 flow
+ *
+ */
+const http = require('http');
+let localServer;
+let localServerSockets;
+
+// We have to kill every socket alive to be able to close the server
+function shutdownLocalServer () {
+  for (const socket of localServerSockets) {
+    socket.destroy();
+  }
+  localServer.close(() => log.debug("Local server closed"));
+}
+
+ipcMain.on('startLocalServer', (event, apiHostName) => {
+  let error;
+  log.debug("Local server started, listening for Oauth2 redirect URI...");
+
+  localServer = http.createServer()
+    .setTimeout(10000)
+    .listen(1612);
+  localServerSockets = new Set();
+
+
+  localServer.on('connection', socket => {
+    localServerSockets.add(socket);
+    socket.once('close', () => {
+      localServerSockets.delete(socket);
+    })
+  });
+
+  localServer.on('request',(req, res) => {
+    log.debug("Oauth2 flow [2]: request received by local server");
+    const parsedUrl = new URL(req.url, `http://${req.headers.host}`);
+
+    if (parsedUrl.pathname === '/oauth2/redirect') {
+      if (parsedUrl.searchParams.has('code')) {
+        log.debug("Oauth2 flow [2]: OK, redirect URI properly formatted with authorization code");
```

Comment:
If you use `state` in the authorization request, this is where you should check that the state has been passed back. (see next comment).

Reply:
89d6ee3568f79c926ce394afe09e2fbfbc03f6c9 Remove state as it would require overly complicated logic to check this state from main process (where the local server run).